File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []
        for point in points:
            dist = sqrt((point[0])**2 + (point[1])**2)
            heapq.heappush(heap,(dist,point))
        ans = []
        for _ in range(k):
            ans.append(heapq.heappop(heap)[1])
        return ans

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
    Queue<int[]> pq = new PriorityQueue<>((a,b) -> (a[0] - b[0]));
    
    
    for(int[] point : points){
        int[] entry = new int[]{distance(point[0], point[1]), point[0], point[1]};
        pq.offer(entry);
        
        
    }
    
    int[][] result  = new int[k][2];
    
    for(int i = 0; i < k; i++){
        
        int[] curr = pq.poll();
        result[i][0] = curr[1];
        result[i][1] = curr[2];
    }
    
    return result;
    
}

public int distance(int x, int y){
    return x*x + y*y;

}
}